ocaml/message-switch/core/mtest.ml

```diff
@@ -0,0 +1,40 @@
+module type MTEST = sig
+  type +'a io
+
+  val mutex_provides_mutal_exclusion : unit -> unit io
+end
+
+module Make =
+functor
+  (M : S.BACKEND)
+  ->
+  struct
+    open M.IO
+
+    type 'a io = 'a M.IO.t
+
+    let ocaml_lock = Mutex.create ()
+
+    let mu = M.Mutex.create ()
+
+    let cond = M.Condition.create ()
+
+    let broadcast () = M.Condition.broadcast cond ()
+
+    let mutex_provides_mutal_exclusion () : unit io =
+      ignore @@ M.Clock.run_after 1 broadcast ;
```

Comment:
Move this after the array creation: then you don't need to sleep for 1s, you can immediately broadcast after the array has been created.